### PR TITLE
Better separate between glue and cli in docs

### PR DESCRIPTION
**Pull request by mdellweg**
The architecture page should properly introduce all sections of the codebase and state their respective purposes.

[noissue]

Review Checklist:
- [ ] An issue is properly linked. [feature and bugfix only]
- [ ] Tests are present or not feasible.
- [ ] Commits are split in a logical way (not historically).

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -1,7 +1,42 @@
-## Architecture
+# Architecture
 
 The Pulp CLI architecture is described in this section.
 
+## Pulp Glue
+
+Pulp CLI provides the `pulp-glue` library as an abstraction layer that let's you perform high level operations in pulp.
```
Comment (ggainey) [inline docs/architecture.md:7]:
```suggestion
Pulp CLI provides the `pulp-glue` library as an abstraction layer that lets you perform high-level operations in pulp.
```

File: docs/architecture.md [hunk 2]
```diff
@@ -1,7 +1,42 @@
...
+#### Deferred Api and Entity lookup
+
+In order to be able to access every (sub-)commands help page,
+it is necessary that no code outside of the final performing command callback accesses the `api` property of the `PulpContext`.
+There are some facilities that perform deferred loading to help with that requirement.
+Those include:
+
+  - `PulpContext.api`: When accessed, the `api.json` file for the addressed server will be read or downloaded and processed.
+    Scheduled version checks will be reevaluated.
+  - `PulpContext.needs_version`: This function can be used at any time to declear that an operation needs a plugin in a version range.
```
Comment (ggainey) [inline docs/architecture.md:31]:
```suggestion
  - `PulpContext.needs_version`: This function can be used at any time to declare that an operation needs a plugin in a version range.
```

File: docs/architecture.md [hunk 3]
```diff
@@ -1,7 +1,42 @@
...
+    The actual check will be performed, once `api` was accessed for the first time, or immediately afterwards.
+  - `PulpEntityContext.entity`: This property can be used to collect lookup attributes for entities by assigining dicts to it.
```
Comment (ggainey) [inline docs/architecture.md:33]:
```suggestion
  - `PulpEntityContext.entity`: This property can be used to collect lookup attributes for entities by assigning dicts to it.
```

File: docs/architecture.md [hunk 4]
```diff
@@ -1,7 +1,42 @@
...
+it is necessary that no code outside of the final performing command callback accesses the `api` property of the `PulpContext`.
+There are some facilities that perform deferred loading to help with that requirement.
+Those include:
+
+  - `PulpContext.api`: When accessed, the `api.json` file for the addressed server will be read or downloaded and processed.
+    Scheduled version checks will be reevaluated.
+  - `PulpContext.needs_version`: This function can be used at any time to declare that an operation needs a plugin in a version range.
+    The actual check will be performed when `api` was accessed for the first time, or immediately afterwards.
+  - `PulpEntityContext.entity`: This property can be used to collect lookup attributes for entities by assigning dicts to it.
+    On read access, the entity lookup will be performed though the `api` property.
```
Comment (ggainey) [inline docs/architecture.md:34]:
```suggestion
    On read access, the entity lookup will be performed through the `api` property.
```

**Review by ggainey — CHANGES_REQUESTED** [review]
Found one last(?) typo I missed prior, sorry

**Review by ggainey — APPROVED** [review]

**Comment by mdellweg** [comment]
> Found one last(?) typo I missed prior, sorry

No problem. T'was me to leave so many around for finding.

**Pull request merged**